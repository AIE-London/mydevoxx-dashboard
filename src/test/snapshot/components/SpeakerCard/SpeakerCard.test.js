import React from "react";
import ReactDOM from "react-dom";
import Speaker from "../../../../main/components/SpeakerCard";
import renderer from "react-test-renderer";

const speakerDetail = {
  name: "Test Speaker",
  company: "Capgemini",
  blog: "personalblog.com",
  talks: [
    "Intro to Devoxx (Room 1 - 11:45)",
    "Intro to Devoxx 2 (Room 2 - 13:45)"
  ]
};

test("SpeakerCard component", () => {
  it("renders without crashing", () => {
    const div = document.createElement("div");
    ReactDOM.render(
      <Speaker
        name="Testy MacTest"
        company="Testy Inc."
        blog="forsomereasonihaveablog.com"
        talks={[
          "The Importance of Jazz in modern society",
          "Rory's Roaring Apps: An M.Night Shayamalan production",
          "That other talk"
        ]}
      />,
      div
    );
  });
});

test("SpeakerCard component with title snapshot", () => {
  const tree = renderer.create(<Speaker speaker={speakerDetail} />).toJSON();
  expect(tree).toMatchSnapshot();
});
