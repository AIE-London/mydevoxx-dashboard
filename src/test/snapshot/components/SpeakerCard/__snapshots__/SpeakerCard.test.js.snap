exports[`test SpeakerCard component with title snapshot 1`] = `
<div
  className="lluTUT row">
  <div
    className="euoSBq">
    <div
      className="row">
      <div
        className="">
        <img
          className="fOjFzA"
          src="defaultIcon.png" />
      </div>
      <div
        className="col-xs">
        <div>
          <h3>
            Testy
          </h3>
          <p>
            Testy Inc.
          </p>
        </div>
      </div>
    </div>
    <div
      className="row">
      <div
        className="">
        <div>
          Personal Blog:
        </div>
        <div
          className="fYLUow">
          blogname.com
        </div>
      </div>
    </div>
    <div
      className="row">
      <div
        className="">
        <div>
          Other Talks:
        </div>
        <ul>
          <div
            className="fYLUow">
            <li>
              Talk 1 is One
            </li>
            <li>
              Talk 2 is Two
            </li>
            <li>
              Talk 3 is Three
            </li>
          </div>
        </ul>
      </div>
    </div>
  </div>
</div>
`;
